File: leetcode.com/python/56_Merge_Intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        if len(intervals) <= 1:
            return intervals
        intervals = sorted(intervals, key = lambda x: x[0])
        mergedIntervals = []
        for i in range(len(intervals)):
            currentInterval = intervals[i]
            if mergedIntervals and currentInterval[0] <= mergedIntervals[-1][1]:
                mergedIntervals[-1][1] = max(mergedIntervals[-1][1], currentInterval[1])
            else:
                mergedIntervals.append(currentInterval)
        return mergedIntervals



sol = Solution()
input = [[1,4],[0,4]]
output = sol.merge(input)
print('Res: ', output)

File: leetcode.com/swift/DS_ALGO_PRAC.playground/Sources/swift/56_Merge_Intervals.swift
import Foundation

class Solution {
    func merge(_ intervals: [[Int]]) -> [[Int]] {
        guard intervals.count > 1 else {
            return intervals
        }
        var sortedIntervals: [[Int]] = intervals.sorted { $0[0] < $1[0] } // Sort the intervals by start time
        print(sortedIntervals)
        var mergedIntervals = [[Int]]()
        mergedIntervals.append(sortedIntervals.first!)
        print(mergedIntervals)
        for index in 1..<sortedIntervals.count {
            var previousInterval = mergedIntervals.last!
            var currentInterval = sortedIntervals[index]
            if previousInterval[1] >= currentInterval[0] {
                mergedIntervals.popLast()
                mergedIntervals.append([min(previousInterval[0], currentInterval[0]), max(previousInterval[1], currentInterval[1])])
            } else {
                mergedIntervals.append(currentInterval)
            }
        }
        return mergedIntervals
    }
}
